prelude/diagnostic.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable global-require */
+/* eslint-disable no-console */
+
+'use strict';
+
+(function installDiagnostic() {
+  const fs = require('fs');
+  const path = require('path');
+
+  const win32 = process.platform === 'win32';
+
+  function dumpLevel(folderPath, level) {
+    let totalSize = 0;
+    const d = fs.readdirSync(folderPath);
+    for (let j = 0; j < d.length; j += 1) {
+      const f = path.join(folderPath, d[j]);
+      //  const isSymbolicLink = fs.statSync(f).isSymbolicLink();
+      const realPath = fs.realpathSync(f);
+      const isSymbolicLink2 = f !== realPath;
+
+      const s = fs.statSync(f);
+      totalSize += s.size;
+      console.log(
+        ' '.padStart(level * 2, ' '),
+        d[j],
+        s.size,
+        isSymbolicLink2 ? `=> ${realPath}` : ' '
+      );
+
+      if (s.isDirectory() && !isSymbolicLink2) {
+        totalSize += dumpLevel(f, level + 1);
+      }
+    }
+    return totalSize;
+  }
+  function wrap(obj, name) {
+    const f = fs[name];
+    obj[name] = (...args) => {
+      const args1 = Object.values(args);
+      console.log(
+        `fs.${name}`,
+        args1.filter((x) => typeof x === 'string')
+      );
+      return f.apply(this, args1);
+    };
+  }
+  if (process.env.DEBUG_PKG) {
+    console.log('------------------------------- virtual file system');
+    const startFolder = win32 ? 'C:\\snapshot' : '/snapshot';
+    const totalSize = dumpLevel(startFolder, 2);
+    console.log('Total size = ', totalSize);
+    wrap(fs, 'openSync');
```

Comment:
What about iterating the prototype keys instead?

Reply:
No all are wrapped